cow_instrument/InstrumentTree.cpp

```diff
@@ -97,15 +100,20 @@ InstrumentTree InstrumentTree::modify(size_t node,
                       currentChildren.end());
     }
   }
-  return InstrumentTree(std::move(newNodes), m_detectorVec.size());
+  return std::unique_ptr<const InstrumentTree>(
+      new InstrumentTree(std::move(newNodes), m_detectorVec.size()));
 }
 
-InstrumentTree InstrumentTree::modify(const Node *node,
-                                      const Command &command) const {
+std::unique_ptr<const InstrumentTree>
```

Comment:
Since `InstrumentTree` has only `const` methods I would use `std::unique_ptr<InstrumentTree>` in all cases. Double enforcing const-ness maybe only adds confusion?